src/web/demo.rkt

```diff
@@ -367,21 +377,27 @@
      (redirect-to (add-prefix (format "~a.~a/graph.html" hash *herbie-commit*)) see-other))
    (url main)))
 
+(define (run-sample hash formula)
+  (hash-set! *jobs* hash (*timeline*))
+  (define sema (make-semaphore))
+  (thread-send *worker-thread* (list 'sample hash formula sema))
+  sema)
+
 ; /api/sample endpoint: test in console on demo page:
 ;; (await fetch('/api/sample', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", seed: 5})})).json()
 (define sample-endpoint
   (post-with-json-response
     (lambda (post-data)
-      (define formula (read-syntax 'web (open-input-string (hash-ref post-data 'formula))))
+      (define formula-str (hash-ref post-data 'formula))
+      (define formula (read-syntax 'web (open-input-string formula-str)))
+      (define hash (sha1 (open-input-string formula-str)))
+      ;; Hmm how should I pass seed?
+      ;; is it ok to use the seed set in the thread?
       (define seed (hash-ref post-data 'seed))
-      (eprintf "Sampling job started on ~a..." formula)
-
-      (define test (parse-test formula))
-      (define result (run-herbie 'sample test #:seed seed #:profile? #f #:timeline-disabled? #t))
-      (define pctx (job-result-backend result))
-
-      (eprintf " complete\n")
-      (hasheq 'points (pcontext->json pctx (context-repr (test-context test)))))))
+      ;; Is this ok because we are multithreaded now?
+      (set-seed! seed)
```

Comment:
This won't work now, because the seed on the worker thread won't be the seed on the main thread. You have to send it in the message.

Reply:
Fixed in next PR.